syntax = "proto3";

package QuantumGUI;

message PolarizationMessage {
	int32 RegisterAddress = 1;
	float Angle = 2;
}

message RegisterAddressMessage {
	int32 RegisterAddress = 1;
}

message ResultMessage {
	int32 RegisterAddress = 1;
}

message VoidMessage {
}

message RegisterStatusMessage {
	repeated NodeStatusMessage nodes = 1;
}

message NodeStatusMessage {
	int32 NodeID = 1;
	float NodeProbability = 2;
}

message FindAngleMessage {
	float pZero = 1;
	float pOne = 2;
}

message AngleMessage {
	float angle = 1;
}

service ControllerView {
	rpc Polarize(PolarizationMessage) returns (VoidMessage) {}
	rpc Measure(RegisterAddressMessage) returns (ResultMessage) {}
	rpc GetRegisterStatus(VoidMessage) returns (RegisterStatusMessage) {}
	rpc FindAngle(FindAngleMessage) returns (AngleMessage) {}
}
